common/OpTestSystem.py

```diff
@@ -1038,14 +1187,80 @@ def sys_ipmi_boot_system_to_petitboot(self, i_console):
         self.cv_IPMI.ipmi_set_boot_to_petitboot()
         self.cv_IPMI.ipmi_power_on()
 
+        self.wait_for_petitboot()
+        self.petitboot_exit_to_shell()
+
+    def wait_for_petitboot(self):
+        console = self.console.get_console()
+        # Wait for petitboot
+        console.expect('Petitboot', timeout=BMC_CONST.PETITBOOT_TIMEOUT)
```

Comment:
Ahh yes, I'd like to extend it to grab logs (or at least save what is from console), set state to UNKNOWN and then throw an exception.

This *could* enable us to write tests that *intentionally* fail to boot/reboot/reach petitboot. e.g. test what happens when we gard out enough hardware that we could no longer boot